src/Microsoft.CodeAnalysis.Razor.Workspaces/WorkspaceMonitor.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    public class WorkspaceMonitor
+    {
+        private readonly Workspace _workspace;
+
+        public WorkspaceMonitor(Workspace workspace)
```

Comment:
In general things in 'workspaces' are going to use MEF and also likely the factory pattern. See https://github.com/dotnet/roslyn/blob/master/src/Workspaces/Core/Portable/Workspace/Host/Mef/ILanguageServiceFactory.cs

Assume that everything here **requires** a separation between abstraction and implementation.